File: CS/LayoutViewCustomButtonContent/LayoutViewCustomButtonHandler.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using DevExpress.XtraGrid.Views.Layout.Handler;
using DevExpress.XtraGrid.Views.Layout;
using DevExpress.XtraGrid.Views.Layout.ViewInfo;
using DevExpress.XtraGrid.Views.Base.ViewInfo;

namespace CustomButtonsLayoutView
{
    class LayoutViewCustomButtonHandler : LayoutViewHandler
    {
        public LayoutViewCustomButtonHandler(LayoutView gridView) : base(gridView) { }

        public override void DoClickAction(BaseHitInfo hitInfo)
        {
            base.DoClickAction(hitInfo);
            LayoutViewCustomButton currentView = View as LayoutViewCustomButton;
            LayoutViewHitInfo currentHitInfo = hitInfo as LayoutViewHitInfo;
            currentView.ButtonsCollection.CheckForCustomButtonClick((int)currentHitInfo.HitTest);
        }
    }
}

File: VB/LayoutViewCustomButtonContent/LayoutViewCustomButtonHandler.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports DevExpress.XtraGrid.Views.Layout.Handler
Imports DevExpress.XtraGrid.Views.Layout
Imports DevExpress.XtraGrid.Views.Layout.ViewInfo
Imports DevExpress.XtraGrid.Views.Base.ViewInfo

Namespace CustomButtonsLayoutView
    Friend Class LayoutViewCustomButtonHandler
        Inherits LayoutViewHandler

        Public Sub New(ByVal gridView As LayoutView)
            MyBase.New(gridView)
        End Sub

        Public Overrides Sub DoClickAction(ByVal hitInfo As BaseHitInfo)
            MyBase.DoClickAction(hitInfo)
            Dim currentView As LayoutViewCustomButton = TryCast(View, LayoutViewCustomButton)
            Dim currentHitInfo As LayoutViewHitInfo = TryCast(hitInfo, LayoutViewHitInfo)
            currentView.ButtonsCollection.CheckForCustomButtonClick(CInt((currentHitInfo.HitTest)))
        End Sub
    End Class
End Namespace
